R/cli_messages.R

```diff
@@ -0,0 +1,62 @@
+#' Check if the data argument is an object of class `FIMSFrame`
+#'
+#' @description
+#' This function checks whether the provided `data` argument is an object of the
+#' class `FIMSFrame`. It ensures that the data adheres to the required structure
+#' for use with FIMS-related functions.
+#' @param data The object to check.
+#' @return NULL If the check passes (i.e., `data` is of class `FIMSFrame`).
+#' The function throws an error if the argument is not of class `FIMSFrame`.
+#' @examples
+#' \dontrun{
+#' data("data_mile1")
+#' fims_data <- FIMSFrame(data_mile1)
+#' check_arg_data(fims_data)
+#' check_arg_data(data = 1:10)
+#' }
+#' @noRd
+check_arg_data <- function(data) {
```

Comment:
After reading a lot of documentation on R functions, we equally call the input to a function a parameter and an argument. So, I am fine with keeping it as "arg" but I still think that it should be `check_data_arg` rather than `check_arg_data`.

Reply:
It looks like a parameter is a variable defined during function definition, while an argument is a value passed to the function when it’s called, from a broader programming perspective. `check_data_arg` works for me. I was considering `check_arg_*` as a prefix in case we write more similar functions, but we can address that later since these are developer-specific functions. I agree that "data" should be renamed to "x".

Reply:
I found out how to use the S4 class to do this rather than having our own check.
```
setMethod(
  "get_data",
  "data.frame",
  function(x) FIMSFrame(x)@data
)
```
This way, when someone passes a data frame to `get_data()` it tries to create a FIMSFrame object, which leads to the validity check for FIMSFrame to be called automatically and users get the following in return
```
> get_data(data.frame(1:10))
Error in FIMSFrame(x) : 
  Check the columns of your data, the following are missing:
data must contain 'type'
data must contain 'name'
data must contain 'datestart'
data must contain 'dateend'
data must contain 'value'
data must contain 'unit'
data must contain 'uncertainty'
data must contain 'age'
```
So, I will just ensure that whenever data is used that an accessor is used first, which will try to make a data.frame into a FIMSFrame object.